earthaccess/auth.py

```diff
@@ -261,9 +301,9 @@ def _netrc(self) -> bool:
             raise FileNotFoundError(f"No .netrc found in {Path.home()}") from err
         except NetrcParseError as err:
             raise NetrcParseError("Unable to parse .netrc") from err
-        if my_netrc["urs.earthdata.nasa.gov"] is not None:
-            username = my_netrc["urs.earthdata.nasa.gov"]["login"]
-            password = my_netrc["urs.earthdata.nasa.gov"]["password"]
+        if my_netrc[self.earthdata_environment["edl"]] is not None:
+            username = my_netrc[self.earthdata_environment["edl"]]["login"]
+            password = my_netrc[self.earthdata_environment["edl"]]["password"]
```

Comment:
This might change to the following, if "system" is used:

```suggestion
        if (creds := my_netrc[self.system.edl_hostname]) is not None:
            username = creds["login"]
            password = creds["password"]
```

Reply:
Nice! this would be cleaner!